lecture3/main.c

```diff
@@ -0,0 +1,193 @@
+
+	if(PutQueue(309) == 1){printf("  Put 309 in queue\n");}
+	else{printf("  Cannot put 309 in queue!\n");}
+
+	if(PutQueue(1235) == 1){printf("  Put 1235 in queue\n");}
+	else{printf("  Cannot put 1235 in queue!\n");}
+
+	if(GetQueue(&digit) == 1){printf("  Get %d from queue\n",digit);}
+	else{printf("  Cannot get digit from queue!\n");}
+
+	if(ReadQueue(&digit) == 1){printf("  Read %d from queue\n",digit);}
+	else{printf("  Cannot read digit from queue!\n");}
+
+	if(PutQueue(5550) == 1){printf("  Put 5550 in queue\n");}
+	else{printf("  Cannot put 5550 in queue!\n");}
+
+	if(PutQueue(10) == 1){printf("  Put 10 in queue\n");}
+	else{printf("  Cannot put 10 in queue!\n");}
+
+	if(GetQueue(&digit) == 1){printf("  Get %d from queue\n",digit);}
+	else{printf("  Cannot get digit from queue!\n");}
+
+	if(PutQueue(-100) == 1){printf("  Put -100 in queue\n");}
+	else{printf("  Cannot put -100 in queue!\n");}
+
+	if(ReadQueue(&digit) == 1){printf("  Read %d from queue\n",digit);}
+	else{printf("  Cannot read digit from queue!\n");}
+
+	if(PutQueue(100) == 1){printf("  Put 100 in queue\n");}
+	else{printf("  Cannot put 100 in queue!\n");}
+
+	if(PutQueue(300) == 1){printf("  Put 300 in queue\n");}
+	else{printf("  Cannot put 300 in queue!\n");}
+
+	if(PutQueue(880) == 1){printf("  Put 880 in queue\n");}
+	else{printf("  Cannot put 880 in queue!\n");}
+
+	if(PutQueue(222) == 1){printf("  Put 222 in queue\n");}
+	else{printf("  Cannot put 222 in queue!\n");}
+
+	if(ReadQueue(&digit) == 1){printf("  Read %d from queue\n",digit);}
+	else{printf("  Cannot read digit from queue!\n");}
+
+	if(GetQueue(&digit) == 1){printf("  Get %d from queue\n",digit);}
+	else{printf("  Cannot get digit from queue!\n");}
+
+	if(PutQueue(321) == 1){printf("  Put 321 in queue\n");}
+	else{printf("  Cannot put 321 in queue!\n");}
+
+	if(PutQueue(0) == 1){printf("  Put 0 in queue\n");}
+	else{printf("  Cannot put 0 in queue!\n");}
+
+	if(ReadQueue(&digit) == 1){printf("  Read %d from queue\n",digit);}
+	else{printf("  Cannot read digit from queue!\n");}
+
+	if(ReadQueue(&digit) == 1){printf("  Read %d from queue\n",digit);}
+	else{printf("  Cannot read digit from queue!\n");}
+
+	if(PutQueue(-57) == 1){printf("  Put -57 in queue\n");}
+	else{printf("  Cannot put -57 in queue!\n");}
+
+	if(PutQueue(900) == 1){printf("  Put 900 in queue\n");}
+	else{printf("  Cannot put 900 in queue!\n");}
+
+	if(PutQueue(202) == 1){printf("  Put 202 in queue\n");}
+	else{printf("  Cannot put 202 in queue!\n");}
+
+	if(GetQueue(&digit) == 1){printf("  Get %d from queue\n",digit);}
+	else{printf("  Cannot get digit from queue!\n");}
+
+	if(PutQueue(9) == 1){printf("  Put 9 in queue\n");}
+	else{printf("  Cannot put 9 in queue!\n");}
+
+	if(ReadQueue(&digit) == 1){printf("  Read %d from queue\n",digit);}
+	else{printf("  Cannot read digit from queue!\n");}
+
+	if(PutQueue(1110) == 1){printf("  Put 1110 in queue\n");}
+	else{printf("  Cannot put 1110 in queue!\n");}
```

Comment:
I would rather have the possibility to add and remove values from stack/queue by myself, instead of just checking through a lot of printed lines of code.

Reply:
Another option would be printing out the current state of the stack/queue after each operation, to make it clear what is happening.